contrib/babelfishpg_tsql/src/pltsql_coerce.c

```diff
@@ -1307,7 +1332,15 @@ tsql_func_select_candidate(List *names,
 	bool			  candidates_are_opers = false;
 
 	if (is_special)
+	{
+		/*
+		 * In case of dump restore we don't require special handling as PG handling will be sufficient
```

Comment:
### 1. Upgrade from Previous version to Current version
-   In this case, Dump would take place according to previous definitions, and during restore PG handling will first try to match argument datatypes, but in this case since one of the earlier definitions is defined with ANYELEMENT, so it will not be able to match the exact definitions and will hit our handling, but we don't need that rather PG handling will take care.

### 2. Upgrade from Current version to future versions
-   In this case, DDLs will be dumped according to current definitions and during restore PG handling will take early exit as it would be able to match with exact definitions, so anyway it will not hit our additional handling

Also updated the description